infra/docker/server.Dockerfile

```
@@ -1,48 +1,38 @@
-# Install all dependencies (cached if package files unchanged)
 RUN pnpm install --frozen-lockfile
 
-# Copy source code
+# Build
+FROM base AS builder
+COPY --from=deps /app/node_modules ./node_modules
+COPY --from=deps /app/server/node_modules ./server/node_modules
+COPY --from=deps /app/shared/types/node_modules ./shared/types/node_modules
 COPY . .
+RUN pnpm --filter @shared/types build
+RUN pnpm --filter server build
```

Hadolint (2.14.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)